### PR TITLE
Markdown show methods for our types

### LINKED ISSUE #273: show methods for other MIME types
- [x] HTML
    - [x] `LinearMixedModel`
    - [x] `GeneralizedLinearMixedModel`
    - [x] `BlockDescription`
    - [ ] ~`PCA`~
    - [x] `LikelihoodRatioTest`
    - [x] `OptSummary`

- [x] Markdown
    - [x] `LinearMixedModel`
    - [x] `GeneralizedLinearMixedModel`
    - [x] `BlockDescription`
    - [ ] ~`PCA`~
    - [x] `LikelihoodRatioTest`
    - [x] `OptSummary`

- [x] LaTeX (maybe)
    - [x] `LinearMixedModel`
    - [x] `GeneralizedLinearMixedModel`
    - [x] `BlockDescription`
    - [ ] ~`PCA`~
    - [x] `LikelihoodRatioTest`
    - [x] `OptSummary`

~Just doing Markdown for now because the other two can be generated from Markdown via pandoc.~

For the `MixedModel`s, we might want to consider a full summary output and a compact output (see  [`lmerOut`](https://bitbucket.org/palday/lmerout) for `pprint.merMod` and `pprint.summary.merMod` for something similar).

### PR DESCRIPTION
**Pull request by palday**
Closes #273. 

~Needs tests.~

I've marked this for v3.2 in the NEWS file and will do the necessary backport as soon as this is merged.

There's no method for `CoefTable` because that would be type piracy. I'll make an [upstream PR](https://github.com/JuliaStats/StatsBase.jl/pull/664).

The Markdown tables for the models are compact. This is intentional -- I think this is journal/publication friendlier and also makes them true Markdown tables. If we decide to add in LaTeX methods, this will also be convenient. The measures of fit (log likelihood, deviance, AIC, BIC) can be extracted separately and if someone really wants to fight with a journal, then they can assemble the pieces from the `VarCorr` and `CoefTable` methods.

`OptSummary` was low-hanging fruit, so I've included it. `PCA` seems fiddly won't be a single table anyway, so I've omitted it. (I also don't expect most people to include that in published work, so it really didn't seem like the effort.)

## `BlockDescription`

|rows|subj|fixed|
|:--|:--:|:--:|
|36|BlkDiag||
|2|Dense|Dense|

|rows|subj|item|fixed|
|:--|:--:|:--:|:--:|
|316|Diagonal|||
|24|Dense|Diag/Dense||
|6|Dense|Dense|Dense|

## `LikelihoodRatioTest`

||model-dof|deviance|χ²|χ²-dof|P(>χ²)|
|:-|-:|-:|-:|-:|:-|
|reaction ~ 1 + days + (1 \| subj)|4|1794| | | |
|reaction ~ 1 + days + (1 + days \| subj)|6|1752|42|2|<1e-09|

## `MixedModel`

|           |    Est.|      SE|    z|     p|σ_subj  |
|:----------|-------:|-------:|----:|-----:|-------:|
|(Intercept)|251.4051|9.506187|26.45|<1e-99| 36.0121|
|days       | 10.4673|0.801735|13.06|<1e-38|        |
|Residual   |30.89543|||||

|           |    Est.|     SE|    z|     p|σ_subj   |
|:----------|-------:|------:|----:|-----:|--------:|
|(Intercept)|251.4051|6.63226|37.91|<1e-99| 23.78047|
|days       | 10.4673|1.50224| 6.97|<1e-11|  5.71683|
|Residual   |25.59182|||||

|            |      Est.|      SE|    z|     p|σ_subj   |σ_item   |
|:-----------|---------:|-------:|----:|-----:|--------:|--------:|
|(Intercept) | 0.1955555|0.405190| 0.48|0.6294| 1.339766| 0.495305|
|anger       | 0.0575541|0.016758| 3.43|0.0006|         |         |
|gender: M   | 0.3207844|0.191266| 1.68|0.0935|         |         |
|btype: scold|-1.0582595|0.256805|-4.12|<1e-04|         |         |
|btype: shout|-2.1047525|0.258529|-8.14|<1e-15|         |         |
|situ: self  |-1.0549790|0.210303|-5.02|<1e-06|         |         |

## `OptSummary`
| | |
|-|-|
|**Initialization**| |
|Initial parameter vector|[1.0, 0.0, 1.0]|
|Initial objective value|1784.642296192471|
|**Optimizer settings**| |
|Optimizer (from NLopt)|`LN_BOBYQA`|
|`Lower bounds`|[0.0, -Inf, 0.0]|
|`ftol_rel`|1.0e-12|
|`ftol_abs`|1.0e-8|
|`xtol_rel`|0.0|
|`xtol_abs`|[1.0e-10, 1.0e-10, 1.0e-10]|
|`initial_step`|[0.75, 1.0, 0.75]|
|`maxfeval`|-1|
|**Result**| |
|Function evaluations|1|
|Final parameter vector|[0.9292, 0.0182, 0.2226]|
|Final objective value|1751.9393|
|Return code|`FTOL_REACHED`|

## `VarCorr`

|   |Column|Variance|Std.Dev.|Corr.|
|:--|:-----|-------:|-------:|----:|
|subj|(Intercept)|565.51069|23.78047| |
| |days|32.68212|5.71683| +0.08|
|Residual| |654.94145|25.59182|

### CODE REVIEW

File: src/pca.jl [hunk 1]
```diff
@@ -54,19 +54,9 @@ Base.propertynames(pca::PCA, private::Bool = false) = (
...
-                    ndigitsvec=ndigitsvec,
-                    ndigitscum=ndigitscum,
-                    covcor=covcor,
-                    loadings=loadings,
-                    variances=variances,
-                    stddevs=stddevs)
-
-function Base.show(io::IO, pca::PCA;
+Base.show(io::IO, pca::PCA; kwargs...) = Base.show(io, MIME"text/plain"(), pca; kwargs...)
+
```
Comment (dmbates) [inline src/pca.jl:58]: Why is the second argument `MIME"text/plain"()` here and `MIME"text/plain"` in other places?  I see that both cases are mentioned in the documentation for `Base.show` but it is not clear to me what the distinction is.

**Review by dmbates — APPROVED** [review]
This looks fine to me.  Thanks very much for doing this - formatting output is one of my least favorite parts of coding - probably tracing back to needing to write Fortran FORMAT statements in my youth.

File: src/pca.jl (hunk 1 above)
Comment (palday) [reply src/pca.jl:58]: Good question! `MIME"text/plain" ` is a type, so the signature includes that. But we only use it for multiple dispatch, hence no binding. But when calling the function, we need to instantiate that type.

Comment (palday) [reply src/pca.jl:58]: I suppose we could also dispatch on the type itself with `::Type{MIME"text/plain"}` but that gives me a headache.

Comment (dmbates) [reply src/pca.jl:58]: The answer makes sense - it was the sort of thing I was anticipating.  No need to change anything.

**Pull request merged**